services/polymart/polymart-base.js

```diff
@@ -0,0 +1,39 @@
+import Joi from 'joi'
+import { BaseJsonService } from '../index.js'
+
+const resourceSchema = Joi.object({
+  response: Joi.object({
+    resource: Joi.object({
+      price: Joi.number().required(),
+      downloads: Joi.string().required(),
+      reviews: Joi.object({
+        count: Joi.number().required(),
+        stars: Joi.number().required(),
+      }),
+      updates: Joi.object({
+        latest: Joi.object({
+          version: Joi.string().required(),
+        }),
+      }),
+    }),
+  }),
```

Comment:
All the `Joi.object()`s (`response`, `resource`, etc) should be `.required()` too as well as just the values inside them. Otherwise the objects themselves are optional.

Reply:
Ok, thanks for the information will add this too.